Mathlib/Init/Algebra/Order.lean

```diff
@@ -337,4 +356,33 @@ theorem le_imp_le_of_lt_imp_lt {β} [Preorder α] [LinearOrder β]
   {a b : α} {c d : β} (H : d < c → b < a) (h : a ≤ b) : c ≤ d :=
 le_of_not_lt $ λ h' => not_le_of_gt (H h') h
 
+section Ord
+
+theorem compare_lt_iff_lt {a b : α} : (compare a b = .lt) ↔ a < b := by
+  rw [LinearOrder.compare_eq_compareOfLessAndEq, compareOfLessAndEq]
+  split_ifs <;> simp only [*, lt_irrefl]
+
+theorem compare_gt_iff_gt {a b : α} : (compare a b = .gt) ↔ a > b := by
+  rw [LinearOrder.compare_eq_compareOfLessAndEq, compareOfLessAndEq]
+  split_ifs <;> simp only [*, lt_irrefl, not_lt_of_gt]
+  case _ h₁ h₂ =>
+    have h : b < a := lt_trichotomy a b |>.resolve_left h₁ |>.resolve_left h₂
+    exact true_iff_iff.2 h
+
+theorem compare_eq_iff_eq {a b : α} : (compare a b = .eq) ↔ a = b := by
+  rw [LinearOrder.compare_eq_compareOfLessAndEq, compareOfLessAndEq]
+  split_ifs <;> simp only []
+  case _ h   => exact false_iff_iff.2 <| ne_iff_lt_or_gt.2 <| .inl h
+  case _ _ h => exact true_iff_iff.2 h
+  case _ _ h => exact false_iff_iff.2 h
+
+theorem compare_trichotomy {a b : α} :
+    (compare a b = .lt) ∨ (compare a b = .eq) ∨ (compare a b = .gt) :=
+  lt_by_cases a b
+    (fun h ↦ Or.inl (compare_lt_iff_lt.2 h))
+    (fun h ↦ Or.inr <| Or.inl (compare_eq_iff_eq.2 h))
+    (fun h ↦ Or.inr <| Or.inr (compare_gt_iff_gt.2 h))
```

Comment:
What;s the point of this lemma? It's true on any `Ord` instance, not just linear orders, as these are the only values of `Ordering` that exist.

Reply:
Whoops! Over-generalized without thinking... (Taking a break now, will fix later)